File: 3747-maximum-difference-between-adjacent-elements-in-a-circular-array/3747-maximum-difference-between-adjacent-elements-in-a-circular-array.java
class Solution {
    public int maxAdjacentDistance(int[] nums) {
        int n = nums.length;
        int ans = 0;
        for (int i = 0; i < n; i++) {
            //Use % n for wrap around and if index is negative the add kro n to make
            //Positive and then wrap around kro
            int lsum = Math.abs(nums[i % n] - nums[(i - 1 + n) % n]);
            int rsum = Math.abs(nums[i % n] - nums[(i + 1) % n]);

            ans = Math.max(ans, Math.max(lsum, rsum));
        }
        return ans;
    }
}

File: 3747-maximum-difference-between-adjacent-elements-in-a-circular-array/3747-maximum-difference-between-adjacent-elements-in-a-circular-array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxAdjacentDistance = function (n) {
    const s = n.length
    let ans = Math.abs(n[0] - n[s - 1])

    for (let i = 0; i < s - 1; i++) {
        ans = Math.max(ans, Math.abs(n[i] - n[i + 1]))
    }
    return ans
};